core/src/main/java/com/linecorp/armeria/common/HttpResponseBuilder.java

```diff
@@ -0,0 +1,270 @@
+        return (HttpResponseBuilder) super.content(contentType, content);
+    }
+
+    /**
+     * Sets the {@link Publisher} for this response.
+     */
+    @Override
+    public HttpResponseBuilder content(MediaType contentType, Publisher<? extends HttpData> content) {
+        return (HttpResponseBuilder) super.content(contentType, content);
+    }
+
+    /**
+     * Sets the content for this response. The {@code content} that is converted into JSON
+     * using the default {@link ObjectMapper}.
+     */
+    @Override
+    public HttpResponseBuilder contentJson(Object content) {
+        return (HttpResponseBuilder) super.contentJson(content);
+    }
+
+    /**
+     * Sets a header for this response. For example:
+     * <pre>{@code
+     * HttpResponse.builder()
+     *             .ok()
+     *             .content("Hello, Armeria")
+     *             .header("Server", "foo")
+     *             .build();
+     * }</pre>
+     */
+    @Override
+    public HttpResponseBuilder header(CharSequence name, Object value) {
+        responseHeadersBuilder.addObject(requireNonNull(name, "name"),
+                                         requireNonNull(value, "value"));
+        return (HttpResponseBuilder) super.header(name, value);
+    }
+
+    /**
+     * Sets multiple headers for this response. For example:
+     * <pre>{@code
+     * HttpResponse.builder()
+     *             .ok()
+     *             .content("Hello, Armeria")
+     *             .headers(HttpHeaders.of("x-test-header", "foo", "Server", "baz"))
+     *             .build();
+     * }</pre>
+     * @see HttpHeaders
+     */
+    @Override
+    public HttpResponseBuilder headers(Iterable<? extends Entry<? extends CharSequence, String>> headers) {
+        return (HttpResponseBuilder) super.headers(headers);
+    }
+
+    /**
+     * Sets HTTP trailers for this response.
+     */
+    @Override
+    public HttpResponseBuilder trailers(Iterable<? extends Entry<? extends CharSequence, String>> trailers) {
+        return (HttpResponseBuilder) super.trailers(trailers);
+    }
+
+    /**
+     * Builds the response.
+     */
+    public HttpResponse build() {
+        httpHeaders().build().forEach(entry -> responseHeadersBuilder.set(entry.getKey(), entry.getValue()));
+        final ResponseHeaders responseHeaders = responseHeadersBuilder.build();
+        final HttpHeadersBuilder trailers = httpTrailers();
+        HttpData content = content();
+        final Publisher<? extends HttpData> publisher = publisher();
+        if (publisher == null) {
+            if (content == null) {
+                content = HttpData.empty();
+            }
+            if (trailers == null) {
+                return HttpResponse.of(responseHeaders, content);
+            } else {
+                return HttpResponse.of(responseHeaders, content, trailers.build());
+            }
+        } else {
+            return HttpResponse.of(responseHeaders, publisher);
```

Comment:
I think we should add trailers if it's not null as we do in the request builder.